src/zh/toptoon/src/eu/kanade/tachiyomi/extension/zh/toptoon/Toptoon.kt

```diff
@@ -0,0 +1,193 @@
+            .map {
+                SManga.create().apply {
+                    title = it.meta.title
+                    url = it.url
+                    thumbnail_url = it.thumbnail.url
+                }
+            }
+        return MangasPage(mangas, false)
+    }
+
+    // Details
+
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        title = document.selectFirst("section.infoContent div.title")!!.text()
+        thumbnail_url = document.selectFirst("div.comicThumb img")!!.absUrl("src")
+        author = document.selectFirst("section.infoContent div.etc")!!.text()
+            .substringAfter("作家 : ").substringBefore("|")
+        description = document.selectFirst("div.comic_story div.desc")!!.text()
+        genre = document.selectFirst("section.infoContent div.hashTag")?.text()
+            ?.replace("#", ", ")
+        if (document.selectFirst("div.etc span.comicDayBox") != null) {
+            status = SManga.ONGOING
+        } else if (document.selectFirst("div.hashTag a[href=/search/keyword/79]") != null) {
+            status = SManga.COMPLETED
+        }
+    }
+
+    // Chapters
+
+    override fun chapterListSelector(): String = "section.episode_area ul.list_area li.episodeBox"
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        name = if (element.selectFirst("button.coin, button.gift, button.waitFree") != null) {
+            "\uD83D\uDD12"
+        } else {
+            ""
+        } + element.selectFirst("div.title")!!.text() + " " +
+            element.selectFirst("div.subTitle")!!.text()
+        date_upload = parseDate(element.selectFirst("div.pubDate")?.text() ?: "")
+    }
+    override fun chapterListParse(response: Response): List<SChapter> {
+        if (response.request.url.pathSegments[0].isEmpty()) {
+            throw Exception("请到WebView确认年满18岁")
+        }
+        return super.chapterListParse(response).asReversed()
+    }
+
+    // Pages
+
+    override fun pageListParse(response: Response): List<Page> {
+        val pathSegments = response.request.url.pathSegments
+        if (pathSegments[0].isEmpty()) {
+            throw Exception("请到WebView确认年满18岁")
+        } else if (pathSegments.size < 2 || pathSegments[1] != "epView") {
+            throw Exception("请确认是否已登录解锁")
+        }
+        return super.pageListParse(response)
+    }
+
+    override fun pageListParse(document: Document): List<Page> {
+        val images = document.select("article.epContent section.imgWrap div.cImg img")
+        return images.mapIndexed { index, img ->
+            Page(index, imageUrl = img.absUrl("data-src"))
+        }
+    }
+
+    override fun imageUrlParse(document: Document): String = throw UnsupportedOperationException()
+
+    private inline fun <reified T> Response.parseAs(): T = use {
```

Comment:
Remove this definition to use `parseAs()` from the `core` module. Should also remove the `json` field.